pkg/sypgp/sypgp.go

```diff
@@ -741,136 +746,162 @@ func ExportPubKey(kpath string, armor bool) error {
+		}
+		err = EncryptKey(newEntity, newPass)
+		if err != nil {
+			return err
 		}
-	}
 
+		// Store the private key
+		err = StorePrivKey(newEntity)
+		if err != nil {
+			return err
+		}
+		fmt.Printf("Key with fingerprint %X succesfully added to the keyring\n", entity.PrimaryKey.Fingerprint)
+	} else {
+		fmt.Printf("The key you want to add with fingerprint %X already belongs to the keyring\n", entity.PrimaryKey.Fingerprint)
+	}
 	return nil
 }
 
 // ImportPubKey Will import a public key from a file (kpath).
-func ImportPubKey(kpath string) error {
+func ImportPubKey(entity *openpgp.Entity) error {
+
 	// Load the local public keys as entitylist
 	publicEntityList, err := LoadPubKeyring()
 	if err != nil {
 		return err
 	}
-	// Load the public key as an entitylist
-	pathEntityList, err := LoadKeyringFromFile(kpath)
-	if err != nil {
-		return err
-	}
+
 	// Get local keystore (where the key will be stored)
 	publicFilePath, err := os.OpenFile(PublicPath(), os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0600)
 	if err != nil {
 		return err
 	}
 	defer publicFilePath.Close()
 
-	// Go through the keyring checking for the given fingerprint
-	for _, pathEntity := range pathEntityList {
-		isInStore := false
-		for _, publicEntity := range publicEntityList {
-			if pathEntity.PrimaryKey.KeyId == publicEntity.PrimaryKey.KeyId {
-				isInStore = true
-				// Verify that this key has already been added
-				break
-			}
+	isInStore := false
+	for _, publicEntity := range publicEntityList {
+		if entity.PrimaryKey.KeyId == publicEntity.PrimaryKey.KeyId {
+			isInStore = true
+			// Verify that this key has already been added
+			break
 		}
-		if !isInStore {
-			if err = pathEntity.Serialize(publicFilePath); err != nil {
-				return err
-			}
-			fmt.Printf("Key with fingerprint %X succesfully added to the keyring\n", pathEntity.PrimaryKey.Fingerprint)
-		} else {
-			fmt.Printf("The key you want to add with fingerprint %X already belongs to the keyring\n", pathEntity.PrimaryKey.Fingerprint)
+	}
+	if !isInStore {
+		if err = entity.Serialize(publicFilePath); err != nil {
+			return err
 		}
+		fmt.Printf("Key with fingerprint %X succesfully added to the keyring\n", entity.PrimaryKey.Fingerprint)
+	} else {
+		fmt.Printf("The key you want to add with fingerprint %X already belongs to the keyring\n", entity.PrimaryKey.Fingerprint)
 	}
+
 	return nil
 }
 
+func getTypesFromEntity(path string) []string {
+	var types []string
+	var block *armor.Block
+
+	f, err := os.Open(path)
+	if err != nil {
+		return types
+	}
+	defer f.Close()
+
+	el, err := openpgp.ReadKeyRing(f)
+	if err != nil {
+		// is armored, so need to identify each of the block types and store them
+		re := openpgp.ReformatGPGExportedFile(f)
+		block, err = armor.Decode(re)
+		types = append(types, block.Type)
+	} else {
+		// is not armored, so obtain the types from entitylist
+		for _, pathEntity := range el {
+			if pathEntity.PrivateKey != nil {
+				types = append(types, "PGP PRIVATE KEY BLOCK")
```

Comment:
It may be better to declare `PGP PRIVATE KEY BLOCK`, and `PGP PUBLIC KEY BLOCK` as `const` at the top of the file.